avocado-setup.py

```diff
@@ -275,19 +277,27 @@ def get_repo(repo, basepath, install=False):
     if os.path.isdir(repo_path):
         logger.info("Updating the repo: %s in %s", repo_name, repo_path)
         cmd = "cd %s;git remote update;git merge origin master" % repo_path
+        err_str = "Failed to update %s repository:" % repo_name
         try:
             status, output = commands.getstatusoutput(cmd)
-            logger.debug("Update Repo: %s\n%s", repo_name, output)
+            if status != 0:
+                logger.error("%s %s", err_str, output)
+                sys.exit(1)
+            logger.debug("%s", output)
```

Comment:
Since we are logging output anyway, why not do it at only one place.
Also, it is better to log the err_str in exception too.

Reply:
IIUC, ` logger.error("%s %s", err_str, output)` and ` logger.debug("%s", output)` you call two times, one we exit during error case and do not print the other

will add err_str in exception...

Reply:
@sathnaga exactly

Reply:
@sathnaga I misunderstood your comment.
What I meant was, there is code duplication of logging output

Reply:
yep, I understood that, but they are under conditional statements with different log(error vs debug) levels, one will not execute if other does, and that is intentional too...